Assets/HoloToolkit/Speech/Scripts/TextToSpeechManager.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System;
+using UnityEngine;
+
+#if WINDOWS_UWP
+using Windows.Media.SpeechSynthesis;
+using Windows.Storage.Streams;
+using System.Threading.Tasks;
+#endif
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Enables text to speech using the Windows 10 <see cref="SpeechSynthesizer"/> class.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="SpeechSynthesizer"/> generates speech as a <see cref="SpeechSynthesisStream"/>. 
+    /// This class converts that stream into a Unity <see cref="AudioClip"/> and plays the clip using 
+    /// the <see cref="AudioSource"/> you supply in the inspector. This allows you to position the voice 
+    /// as desired in 3D space. One recommended approach is to place the AudioSource on an empty 
+    /// GameObject that is a child of Main Camera and position it approximately 0.6 units above the 
+    /// camera. This orientation will sound similar to Cortanas speech in the OS.
+    /// </remarks>
+    public class TextToSpeechManager : Singleton<TextToSpeechManager>
```

Comment:
Oh, and just to mention in case it wasn't clear, spatial speech is entirely supported. The speech is played out of whatever AudioSource you associate with the TextToSpeechManager. So you can have as many AudioSources and as many TextToSpeechManagers as you want. And obviously the AudioSources can have their spatial blend set to 100% 3D. So the holoportation scenario you linked to above is certainly possible.